### DISCUSSION TITLE
Add A1111-style CPU-powered RNG/Seed generation and noise_source selector?

### DISCUSSION THREAD

**Discussion by teward** [Comfy-Org/ComfyUI#3333] [Ideas]
In Automatic1111, there was discrepancy when different types of GPUs, etc. were used and trying to produce consistent seeds and outputs.  To that end, A1111 implemented noise generation that utilized NV-like behavior but ultimately was still CPU-generated.

Is there a possibility that this could be implemented, and then a noise source option be added to KSampler of "NV" which would be similar to A1111 and implement nVidia like generation on CPU for seed values if someone chose to?

This way we can emulate nVidia style seeds and tensors in the system, but without needing to rely on CUDA cross-functionality.  This would also allow AMD cards to get consistent seeds and tensors similar to nVidia generated RNG data as well.

(The last time this was discussed was in 2023 and the discussion was "No I'm not going to implement this", but because A1111 has a CPU-powered mechanism to generate nVidia style tensors and RNGs, I was curious if this could be reasked / reevaluated).

**Comment by comfyanonymous** [comment]
You can make a node that outputs a NOISE like this one: https://github.com/comfyanonymous/ComfyUI/blob/master/comfy_extras/nodes_custom_sampler.py#L471

You can then use it with the SamplerCustomAdvanced node.
